crates/sui-graphql-rpc/src/types/validator.rs

```diff
@@ -1,71 +1,230 @@
+    async fn next_epoch_credentials(&self) -> Option<ValidatorCredentials> {
+        let v = &self.validator_summary;
+        let credentials = ValidatorCredentials {
+            protocol_pub_key: v
+                .next_epoch_protocol_pubkey_bytes
+                .as_ref()
+                .map(Base64::from),
+            network_pub_key: v.next_epoch_network_pubkey_bytes.as_ref().map(Base64::from),
+            worker_pub_key: v.next_epoch_worker_pubkey_bytes.as_ref().map(Base64::from),
+            proof_of_possession: v.next_epoch_proof_of_possession.as_ref().map(Base64::from),
+            net_address: v.next_epoch_net_address.clone(),
+            p2p_address: v.next_epoch_p2p_address.clone(),
+            primary_address: v.next_epoch_primary_address.clone(),
+            worker_address: v.next_epoch_worker_address.clone(),
+        };
+        Some(credentials)
+    }
+
+    /// Validator's name.
+    async fn name(&self) -> Option<String> {
+        Some(self.validator_summary.name.clone())
+    }
+
+    /// Validator's description.
+    async fn description(&self) -> Option<String> {
+        Some(self.validator_summary.description.clone())
+    }
+
+    /// Validator's url containing their custom image.
+    async fn image_url(&self) -> Option<String> {
+        Some(self.validator_summary.image_url.clone())
+    }
+
+    /// Validator's homepage URL.
+    async fn project_url(&self) -> Option<String> {
+        Some(self.validator_summary.project_url.clone())
+    }
+
+    /// The validator's current valid `Cap` object's ID
     async fn operation_cap(&self, ctx: &Context<'_>) -> Result<Option<MoveObject>> {
         ctx.data_unchecked::<PgManager>()
-            .fetch_move_obj(self.operation_cap_id, None)
+            .fetch_move_obj(self.operation_cap_id(), None)
             .await
             .extend()
     }
 
+    /// The validator's current staking pool object
     async fn staking_pool(&self, ctx: &Context<'_>) -> Result<Option<MoveObject>> {
         ctx.data_unchecked::<PgManager>()
-            .fetch_move_obj(self.operation_cap_id, None)
+            .fetch_move_obj(self.staking_pool_id(), None)
             .await
             .extend()
     }
 
+    /// The validator's current exchange object
     async fn exchange_rates(&self, ctx: &Context<'_>) -> Result<Option<MoveObject>> {
         ctx.data_unchecked::<PgManager>()
-            .fetch_move_obj(self.operation_cap_id, None)
+            .fetch_move_obj(self.exchange_rates_id(), None)
             .await
             .extend()
     }
+
+    /// Number of exchange rates in the table.
+    async fn exchange_rates_size(&self) -> Option<u64> {
+        Some(self.validator_summary.exchange_rates_size)
+    }
+
+    /// The epoch at which this pool became active.
+    async fn staking_pool_activation_epoch(&self) -> Option<u64> {
+        self.validator_summary.staking_pool_activation_epoch
+    }
+
+    /// The total number of SUI tokens in this pool.
```

Comment:
Number implies a quantity of `Coin<SUI>` objects rather than the total balance, to me.